common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts

```diff
@@ -0,0 +1,72 @@
+import path = require("path");
+import * as tl from 'azure-pipelines-task-lib/task';
+import { getSystemAccessToken } from "./webapi";
+import fetch from "node-fetch";
+
+tl.setResourcePath(path.join(__dirname, 'module.json'), true);
+
+export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string>
+{
+    try {
+        let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
+        tl.debug(tl.loc('Info_UsingTenantId', tenantId));
+        const systemAccessToken = getSystemAccessToken();
+        const url = process.env["SYSTEM_OIDCREQUESTURI"]+"?api-version=7.1&serviceConnectionId="+serviceConnectionName;
+        
+        const ADOResponse: {oidcToken: string} = await (await fetch(url, 
+        {
+            method: 'POST', 
+            headers: 
+            {
+                'Content-Type': 'application/json',
+                'Authorization': 'Bearer '+ systemAccessToken
+            }
+        })).json() as {oidcToken: string};
+
+        tl.setSecret(ADOResponse.oidcToken);
+        let entraURI = "https://login.windows-ppe.net/"+tenant+"/oauth2/v2.0/token";
```

Comment:
nit: just verifying that we should be hitting the PPE endpoint here

Reply:
grr we should not brb

Reply:
fixed